src/context/MassetsProvider.ts

```diff
@@ -21,24 +21,29 @@ const massets: Masset[] = [
       address: process.env.REACT_APP_MUSD_SAVINGS_ADDRESS as string,
     },
   },
-  // TODO enable mBTC
-  // {
-  //   name: 'mBTC',
-  //   address: process.env.REACT_APP_MBTC_ADDRESS as string,
-  //   savingsContract: {
-  //     address: process.env.REACT_APP_MBTC_SAVINGS_ADDRESS as string,
-  //   },
-  // },
+  {
+    name: 'mBTC',
+    address: process.env.REACT_APP_MBTC_ADDRESS as string,
+    savingsContract: {
+      address: process.env.REACT_APP_MBTC_SAVINGS_ADDRESS as string,
+    },
+  },
 ];
 
+const retrievedMasset = localStorage.getItem('selectedMasset');
+
 const initialState: State = {
-  selectedMasset: massets[0],
   massets,
+  selectedMasset: retrievedMasset ? JSON.parse(retrievedMasset) : massets[0],
```

Comment:
Thinking about it, `selectedMasset` should probably be a maybe type. Defaulting to mUSD might be making things more difficult.

Reply:
I'd recommend either using an initialiser function for the initialState (as per React docs), or an effect that uses the route params as a dependency and dispatches the action.